kinder/pkg/cri/host/archive.go

```diff
@@ -185,6 +199,44 @@ func editManifestRepositories(raw []byte, editRepositories func(string) string)
 	return json.Marshal(entries)
 }
 
+// https://github.com/opencontainers/image-spec/blob/main/image-index.md
+type index struct {
```

Comment:
Editing contents of oci objects in this way seems problematic. https://github.com/opencontainers/image-spec/blob/v1.0/specs-go/v1/index.go#L21 https://github.com/opencontainers/image-spec/blob/v1.0/specs-go/v1/descriptor.go#L55 These types have major changes in oci 1.1 and even 1.0 seems different. Maybe I need more context, on my phone ATM.

Reply:
the thing here is a hack that has to be babysitted anyway if k8s switches to non backwards compatible spec that kinder does not support.

as long as k8s image maintainers tell us  all should be good.

Reply:
Which spec does kinder support? If you wanted to support the 3 valid spec schemas, docker 2, oci 1.0 and 1.1 you could switch on type before doing the conversions. But I'm still not sure why these files are being edited.

Reply:
with this PR oci v2, before it only docker 1.0, 1.1, 1.2.

the k8s image tars from CI are amd-64 only. kinder here is making edits to change them from "-amd64" to "", so that kubeadm can work with these images. kubeadm needs images that are multi-os-arch (manifest lists) without arches/os in the image names.